andela/knights-bn-backend
# #170947675 user can  delete their comment

#### What does this PR do?
- Allow users to delete comments on their request.
#### Description of Task to be completed?
- As the owner of the comment, I should be able to delete unwanted comments
#### How should this be manually tested?
- run `git clone https://github.com/andela/knights-bn-backend.git` to clone the repo.
- run `cd knights-bn-backend` in your terminal to enter into the repository.
- run `git checkout ft-delete-comment-170947675 ` to switch to this branch.
- run `npm install` to install dependencies.

- create a .env file then add environment variable as shown in ` .env.example`,  check our slack channel `code-knights-12` for .env


- create two Postgres database for development and then test
- run `npm run createTables` to create tables for you

- run `npm run dev-start` to start the server.

- login using this user or any other user POST `/api/v1/auth/login`

  { 
       "email": "murenzijames@gmail.com", 
       "password": "Niyonkuru@1"
 }

- make a new request by POST `http://localhost:4000/api/v1/trips/returnTrip`

{
    "origin": "Burundi",
    "destination": "Rwanda",
    "departureDate": "2020-07-20",
    "returnDate": "2020-08-20",
    "accommodation": "XYZ campus",
    "reason": "Maxime test"
}

- make a comment on a comment POST `http://localhost:4000/api/v1/trips/comment?requestId=ID` and include the comment in your request body

{
	"comment": "new comment"
}

- then to the see your comment go to 'localhost:4000/api/v1/trips/myRequest'

- Delete one of the desired comment. `DELETE`

          `http://localhost:4000/api/v1/trips/comment?commentId=id`

- view again your request to see your comment 'localhost:4000/api/v1/trips/myRequest'



#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?

- [#170947675](https://www.pivotaltracker.com/story/show/170947675)

#### Screenshots (if appropriate)
              

Create a request

<img width="935" alt="Screen Shot 2020-03-04 at 16 04 10" src="https://user-images.githubusercontent.com/48021034/75887066-e49b1a80-5e31-11ea-9fe6-a242d254be1a.png">



comment on the request

<img width="851" alt="Screen Shot 2020-03-04 at 16 04 18" src="https://user-images.githubusercontent.com/48021034/75887106-f2e93680-5e31-11ea-84d8-65f1256e1b0b.png">


view all my request with comment

<img width="577" alt="Screen Shot 2020-03-04 at 16 04 41" src="https://user-images.githubusercontent.com/48021034/75887143-00062580-5e32-11ea-8e36-c53224a2cecf.png">

Delete comment

<img width="786" alt="Screen Shot 2020-03-05 at 21 05 26" src="https://user-images.githubusercontent.com/48021034/76016182-104afd00-5f25-11ea-943c-e4f02400bf44.png">

View comment `marked as isVisible: false`

<img width="631" alt="Screen Shot 2020-03-05 at 21 06 19" src="https://user-images.githubusercontent.com/48021034/76016257-2f498f00-5f25-11ea-898d-329b7fcc2770.png">


#### Questions:
N/A
